pkg/controllers/organization/service_proxy_controller.go

```diff
@@ -69,13 +84,23 @@ func (r *ServiceProxyReconciler) reconcileServiceProxy(ctx context.Context, org
 		return fmt.Errorf("failed to marshal version.GitCommit: %w", err)
 	}
 
+	var pluginDefinition = new(greenhousesapv1alpha1.PluginDefinition)
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: serviceProxyName, Namespace: ""}, pluginDefinition); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.FromContext(ctx).Info("plugin definition for service-proxy not found")
+			return nil
+		}
+		log.FromContext(ctx).Info("failed to get plugin definition for service-proxy", "error", err)
+		return nil
```

Comment:
We could add a test for this, right?

Reply:
I've added two tests for ServiceProxyController.